### DISCUSSION TITLE
CDC & MSC & RNDIS at the same time does not work

### DISCUSSION THREAD

**Discussion by tecnovel** [hathach/tinyusb#2597] [Q&A]
Hi, i'm working now over a month on this project and i have a version where cdc and msc is working at the same time and another version where the rndis class is working but the cdc and msc is not working anymore. I tried to analyse the usb descriptors with usblyzer software, but they look right except these 4 descriptors:

Unrecognized Wireless USB Class-Specific Descriptor 
Offset Field Size Value Description 
0 bLength 1 05h  
1 bDescriptorType 1 24h Unrecognized Wireless USB Class-Specific 
2  3 00 10 01   

Unrecognized Wireless USB Class-Specific Descriptor 
Offset Field Size Value Description 
0 bLength 1 05h  
1 bDescriptorType 1 24h Unrecognized Wireless USB Class-Specific 
2  3 01 00 01   

Unrecognized Wireless USB Class-Specific Descriptor 
Offset Field Size Value Description 
0 bLength 1 04h  
1 bDescriptorType 1 24h Unrecognized Wireless USB Class-Specific 
2  2 02 00   

Unrecognized Wireless USB Class-Specific Descriptor 
Offset Field Size Value Description 
0 bLength 1 05h  
1 bDescriptorType 1 24h Unrecognized Wireless USB Class-Specific 
2  3 06 00 01  

but the rndis is working so i think they are not a problem.

I found out, if i change the order in the interface enum:
enum
{
	ITF_NUM_NET,
	ITF_NUM_NET_DATA,
	ITF_NUM_CDC,
	ITF_NUM_CDC_DATA,
	ITF_NUM_MSC,

	ITF_NUM_TOTAL
};

i can make the cdc class work again but the rndis class is not not working. (If i change the position from ITF_NUM_CDC + DATA to position 0).

Does someone know what possible mistakes could be?

Please help me, because it will be a very nice project with the other code around it and i will also share it here. Thanks!

**Comment by ymj-123** [comment]
I had the same problem. Have you solved it?
